packages/@aws-cdk/aws-iam/lib/oidc-provider/external.ts

```diff
@@ -20,26 +21,77 @@ function defaultLogger(fmt: string, ...args: any[]) {
+export async function downloadThumbprint(issuerUrl: string) {
+
+  external.log(`Downloading certificate authority thumbprint for ${issuerUrl}`);
+
   return new Promise<string>((ok, ko) => {
     const purl = url.parse(issuerUrl);
     const port = purl.port ? parseInt(purl.port, 10) : 443;
+
     if (!purl.host) {
       return ko(new Error(`unable to determine host from issuer url ${issuerUrl}`));
     }
+
     const socket = tls.connect(port, purl.host, { rejectUnauthorized: false, servername: purl.host });
     socket.once('error', ko);
+
     socket.once('secureConnect', () => {
-      const cert = socket.getPeerCertificate();
+      // This set to `true` would return the entire chain of certificates as a circular reference object
+      let cert = socket.getPeerCertificate(true);
+
+      const unqiueCerts = new Set<DetailedPeerCertificate>();
+      do {
+        unqiueCerts.add(cert);
+        cert = cert.issuerCertificate;
+      } while ( cert && typeof cert === 'object' && !unqiueCerts.has(cert));
+
+      // The last `cert` obtained must be the root certificate in the certificate chain
+      const rootCert = [...unqiueCerts].pop()!;
+
+      // Add `ca: true` when node merges the feature. Awaiting resolution: https://github.com/nodejs/node/issues/44905
+      if (!(util.isDeepStrictEqual(rootCert.issuer, rootCert.subject))) {
+        return ko(new Error(`Subject and Issuer of certificate received are different. 
+        Received: \'Subject\' is ${JSON.stringify(rootCert.subject, null, 4)} and \'Issuer\':${JSON.stringify(rootCert.issuer, null, 4)}`));
+      }
+
+      const validTo = new Date(rootCert.valid_to);
+      const certificateValidity = getCertificateValidity(validTo);
+
+      if (certificateValidity < 0) {
+        return ko(new Error(`The certificate has already expired on: ${validTo.toUTCString()}`));
+      }
+
+      // Warning user if certificate validity is expiring within 6 months
+      if (certificateValidity < 180) {
+        /* eslint-disable-next-line no-console */
+        console.warn(`The root certificate obtained would expire in ${certificateValidity} days!`);
+      }
+
       socket.end();
-      const thumbprint = cert.fingerprint.split(':').join('');
-      external.log(`certificate authority thumbprint for ${issuerUrl} is ${thumbprint}`);
+
+      const thumbprint = rootCert.fingerprint.split(':').join('');
+      external.log(`Certificate Authority thumbprint for ${issuerUrl} is ${thumbprint}`);
+
       ok(thumbprint);
     });
   });
 }
 
+/**
+ * To get the validity timeline for the certificate
+ * @param certDate The valid to date for the certificate
+ * @returns The number of days the certificate is valid wrt current date
+ */
+function getCertificateValidity(certDate: Date): Number {
+  const millisecondsInDay = 24 * 60 * 60 * 1000;
+  const currentDate = new Date();
+
+  const validity = Math.round((certDate.getTime() - currentDate.getTime()) / millisecondsInDay);
```

Comment:
`Duration` has this sort of thing built in. I'm thinking we should use that here instead of the manual logic?

Reply:
Great, I was not aware of that. Will look into it.

Reply:
I was not able to find how this can be done using Duration. Do you have an example that you can share?

Reply:
It has less built in than I remembered. Feel free to disregard this comment.